src/predictive-text-input/predictive-text-input.tsx

```diff
@@ -132,6 +138,39 @@ export const PredictiveTextInput = <T, V>({
         setIsOptionSelected(!!selectedOption);
     }, [selectedOption]);
 
+    useEffect(() => {
+        const state = getItemsLoadState();
+        const count = options?.length ?? 0;
+
+        if (state === "loading") {
+            setResultAnnouncement("Loading suggested results");
+            return;
+        }
+
+        if (state === "fail") {
+            setResultAnnouncement("Suggestions failed to load. Try again.");
+            return;
+        }
+
+        if (count === 0) {
+            setResultAnnouncement(input ? "No result found." : null);
+        } else {
+            setResultAnnouncement(
+                `${count} result${
+                    count > 1 ? "s" : ""
+                } found. Press down arrow to scroll through the list.`
+            );
+        }
+    }, [options, input, isError, isLoading]);
+
+    useEffect(() => {
+        if (isOpen) {
+            setResultAnnouncement("Listbox, expanded");
+        } else {
+            setResultAnnouncement("Listbox, collapsed");
+        }
```

Comment:
with the above change, this is not needed as `aria-expanded` handles the collapsed state